elementary/monitor/api/alerts/alerts.py

```diff
@@ -146,26 +137,25 @@ def _sort_alerts(
 
         return AlertsQueryResult(
             alerts=filter_alerts(alerts_to_send, filter),
-            malformed_alerts=filter_alerts(malformed_alerts_to_send, filter),
+            malformed_alerts=cast(
+                List[MalformedAlert], filter_alerts(malformed_alerts_to_send, filter)
+            ),
             alerts_to_skip=filter_alerts(alerts_to_skip, filter),
         )
 
+    @staticmethod
     def _get_suppressed_alerts(
-        self,
-        alerts: Union[
-            AlertsQueryResult[TestAlert],
-            AlertsQueryResult[ModelAlert],
-            AlertsQueryResult[SourceFreshnessAlert],
-        ],
+        alerts: AlertsQueryResult[ValidAlertType],
         last_alert_sent_times: Dict[str, str],
     ) -> List[str]:
         suppressed_alerts = []
         current_time_utc = datetime.utcnow()
-        for alert in [*alerts.alerts, *alerts.malformed_alerts]:
-            alert_class_id = alert.alert_class_id
+        all_alerts: List[Alert] = [*alerts.alerts, *alerts.malformed_alerts]
+        for alert in all_alerts:
+            alert_class_id = alert.alert_class_id or ""
```

Comment:
So I did this in a bunch of places
Because in many classes we have fields defined as optional, where practically they are not really optional (or not optional in specific flows)
I wasn't sure if it's safe to change them not to be optional (can cause pydantic to fail in runtime when we initialize classes from dbt results)